docs/url/url.md

```diff
@@ -30,6 +30,15 @@ This document defines semantic conventions that describe URL and its components.
 | `url.path` | string | The [URI path](https://www.rfc-editor.org/rfc/rfc3986#section-3.3) component [2] | `/search` | Recommended |
 | `url.query` | string | The [URI query](https://www.rfc-editor.org/rfc/rfc3986#section-3.4) component [3] | `q=OpenTelemetry` | Recommended |
 | `url.fragment` | string | The [URI fragment](https://www.rfc-editor.org/rfc/rfc3986#section-3.5) component | `SemConv` | Recommended |
+| `url.registered_domain` | string | The highest registered url domain, stripped of the subdomain. [4] | `example.com` | Opt-In |
+| `url.subdomain` | string | The subdomain portion of a fully qualified domain name includes all of the names except the host name under the registered_domain. In a partially qualified domain, or if the the qualification level of the full name cannot be determined, subdomain contains all of the names below the registered domain. [5] | `east` | Opt-In |
+| `url.top_level_domain` | string | The effective top level domain (eTLD), also known as the domain suffix, is the last part of the domain name. For example, the top level domain for example.com is "com". [6] | `co.uk` | Opt-In |
+| `url.username` | string | Username of the request. | `user42` | Opt-In |
+| `url.password` | string | Password of the request. | `changeme` | Opt-In |
+| `url.extension` | string | The field contains the file extension from the original request url, excluding the leading dot. [7] | `png` | Opt-In |
+| `url.domain` | string | Domain of the url, such as `www.opentelemetry.io`. [8] | `www.opentelemetry.io` | Opt-In |
```

Comment:
@Oberon00 Thanks for the review!

I'd like to use your comment to address two aspects here: one that is directly related to this proposal here and one general aspect.

## This `url.*` attributes proposal
While it's true that at a first glance that looks redundant, I think there is reason for existence for both, because of the following:
1. The semantics (or the context) in which these attributes are used can be slightly different. `url.*` fields describe a URL `server/client.address/domain` attributes describe a client-server relationship. There are cases (that ECS users already rely on) in which `url.*` fields are used, including security use cases (e.g. analyzing access logs and their patterns for security patterns).
2. This proposal here only defines attributes that are NOT set into a specific semantic convention context, yet. So they are not part of the HTTP semantic conventions and do not conflict or imply redundancy on the HTTP semantic conventions (where `server.address`, etc. are used).
3. This PR is basically an implementation step of what we agreed on in https://github.com/open-telemetry/oteps/pull/222

## Seeking general agreement
I'd like us to come to a general agreement regarding https://github.com/open-telemetry/oteps/pull/222 on the following:

- When adding ECS fields to semantic attributes that **do not conflict with existing attributes in the context of individual semantic conventions**, let's be less strict on the following aspects, with the goal to move faster with the ECS merger (otherwise it will take very long for us to get ECS in):
  - redundancy arguments
  - syntactical attribute naming rules (e.g. plural vs. singular)
  - naming preferences
  - other "soft" arguments
- I think my proposal https://github.com/open-telemetry/semantic-conventions/issues/197 would help with that as it decouples definition of attributes (which we can do rather uncomplicated) from usage in concrete semantic conventions. 
- Let's consider ECS fields and entire field groups that are not covered by OTel semconv, yet, as "low hanging fruits" for merging ECS into OTel and move fast with those (as there are not conflicts and those are practice-proven).  So, let's try to avoid discussions related to the above-mentioned arguments on each of the PRs we'll have regarding adding ECS fields to OTel. (Note: I'm not implying that we should not have discussions at all in those case where it makes sense, just trying to avoid similar discussions to repeat.)

These are the reasons for the above proposal:
- In https://github.com/open-telemetry/oteps/pull/222 we agreed on merging ECS with OTel, so the faster we do at least the low hanging fruits the better and less friction for the communities (ECS and OTel).
- A big community is relying on ECS fields today, and there are many practice-proven use cases for the existing ECS fields (that will become OTel use cases in the future as well). So whenever possible we should avoid breaking changes (including dropping fields) on both sides (OTel and ECS). The above case is a great example of when it doesn't hurt to add these fields to OTel (because there are no conflicts and we are not proposing to use them in the HTTP semconv, so even redundancy is not a reason). But, for ECS users it would be a huge breaking change to drop the fields.
- The biggest value of https://github.com/open-telemetry/oteps/pull/222 is to merge the ECS and OTel communities into one community with all its mutually additional use cases. However, that only works if we do not build _everything_ from scratch but try to keep (on both ECS and OTel semconv) things that were proven in practice (in case there are not conflicts and breaking changes can be avoided).
- The faster we merge ECS into OTel the less friction there is for the community and the faster we can start expanding OTel with ECS use cases. 

@open-telemetry/specs-semconv-approvers @open-telemetry/specs-semconv-maintainers WDYT about the above general proposal? If it helps I can extract it into a separate issue, or so?

Reply:
> The semantics (or the context) in which these attributes are used can be slightly different.

Emphasis on slightly. There is server.address for the "logical" address and server.socket.domain for the physical address. We will probably also get server.domain. Now this PR proposes a third (fourth) attribute url.domain. I understand that this is sensible for ECS compatibility reasons. Could we mark the attribute as "compatibility-only"? Specific semantic conventions can have (and do have) more specific usage instructions, so I believe there is really no need from semantic convention designer side for this new attribute.

Although you say:

> There are cases (that ECS users already rely on) in which url.* fields are used, including security use cases (e.g. analyzing access logs and their patterns for security patterns) that are different to the server/client.* fields.

I would be interested to know more. It sound like in these cases, what is stored in server.* might be better stored in server.socket.*?